src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs

```diff
@@ -271,8 +271,14 @@ private void ReadSettings(PackageRestoreInputs packageRestoreInputs)
                     GetInstalledPackageReferences(packageReferenceFile, allowDuplicatePackageIds: true));
             }
 
+            // If PackageSaveMode.None is provided, this means that the user did not specify a
+            // PackageSaveMode. In the case of V2 restore, this should default to PackageSaveMode.Defaultv2.
+            var packageSaveMode = EffectivePackageSaveMode == Packaging.PackageSaveMode.None ?
```

Comment:
`EffectivePackageSaveMode` is `None` when `-PackageSaveMode` is excluded. This is already treated as `Defaultv3` for V3 restore and should be treated as `Defaultv2` for V2 restore. This is already case in the actual restore, but not in the preliminary missing packages check (which is this step).